jwst/extract_1d/extract.py

```diff
@@ -2971,8 +2981,13 @@ def do_extract1d(input_model, ref_dict, smoothing_length=None,
                                                            sp_order)
                     else:
                         wcs = input_model.meta.wcs
+
+                    wcs_shape = input_model.data.shape
+                    if len(wcs_shape) > 2:
+                        # CubeModel
+                        wcs_shape = wcs_shape[1:]
                     pixel_solid_angle = util.pixel_area(wcs,
```

Comment:
Hmmm, I just noticed that this section for getting `pixel_solid_angle` is called for every integration.  The value should be the same for every integration, and I have no idea why I didn't limit this to being called only for `integ == 0`.  This is my mistake, and it may explain why some tests have been taking over a day to run.